src/main/java/slimeknights/tconstruct/blocks/GadgetBlocks.java

```diff
@@ -1,78 +1,59 @@
+import slimeknights.tconstruct.library.registration.BlockDeferredRegister;
+import slimeknights.tconstruct.library.registration.ItemDeferredRegister;
+import slimeknights.tconstruct.library.registration.object.BlockItemObject;
+import slimeknights.tconstruct.library.registration.object.ItemObject;
 
-import static slimeknights.tconstruct.common.TinkerPulse.injected;
-
-@ObjectHolder(TConstruct.modID)
 @Mod.EventBusSubscriber(modid = TConstruct.modID, bus = Mod.EventBusSubscriber.Bus.MOD)
 public final class GadgetBlocks {
-
-  public static final LadderBlock stone_ladder = injected();
-  public static final TorchBlock stone_torch = injected();
-  public static final WallTorchBlock wall_stone_torch = injected();
-  public static final PunjiBlock punji = injected();
-  public static final RailBlock wooden_rail = injected();
-  public static final DropperRailBlock wooden_dropper_rail = injected();
-
-  @SubscribeEvent
-  static void registerBlocks(final RegistryEvent.Register<Block> event) {
-    BlockRegistryAdapter registry = new BlockRegistryAdapter(event.getRegistry());
-
-    registry.register(new LadderBlock(BlockProperties.STONE_LADDER) {}, "stone_ladder");
-
-    registry.register(new TorchBlock(BlockProperties.STONE_TORCH) {}, "stone_torch");
-    registry.register(new WallTorchBlock(BlockProperties.STONE_TORCH) {}, "wall_stone_torch");
-
-    registry.register(new RailBlock(BlockProperties.WOODEN_RAIL) {}, "wooden_rail");
-    registry.register(new DropperRailBlock(BlockProperties.WOODEN_RAIL), "wooden_dropper_rail");
-
-    registry.register(new PunjiBlock(BlockProperties.PUNJI), "punji");
+  private static final BlockDeferredRegister BLOCKS = new BlockDeferredRegister(TConstruct.modID);
+  private static final ItemDeferredRegister ITEMS = new ItemDeferredRegister(TConstruct.modID);
+  private static final Item.Properties gadgetProps = new Item.Properties().group(TinkerRegistry.tabGadgets);
+
+  public static void init() {
+    IEventBus modEventBus = FMLJavaModLoadingContext.get().getModEventBus();
+    BLOCKS.register(modEventBus);
+    ITEMS.register(modEventBus);
   }
 
-  @SubscribeEvent
-  static void registerBlockItems(final RegistryEvent.Register<Item> event) {
-    BlockItemRegistryAdapter registry = new BlockItemRegistryAdapter(event.getRegistry(), TinkerRegistry.tabGadgets);
-
-    registry.registerBlockItem(GadgetBlocks.stone_ladder);
-
-    registry.registerBlockItem(new WallOrFloorItem(GadgetBlocks.stone_torch, GadgetBlocks.wall_stone_torch, (new Item.Properties()).group(TinkerRegistry.tabGadgets)));
-
-    registry.registerBlockItem(GadgetBlocks.punji);
-
-    registry.registerBlockItem(GadgetBlocks.wooden_rail);
-    registry.registerBlockItem(GadgetBlocks.wooden_dropper_rail);
-  }
+  public static final BlockItemObject<LadderBlock> stone_ladder = BLOCKS.register("stone_ladder", () -> new LadderBlock(BlockProperties.STONE_LADDER) {}, gadgetProps);
+  // TODO: Need a combo reg for these... two blocks and one item.
```

Comment:
I'd ditch this comment now, we can reconsider the need for a combo in the future if needed, but as it stands we have no other wall/floor items

Reply:
Ah yeah. Missed it.